toolchain/base/value_store.h

```diff
@@ -106,6 +107,7 @@ class ValueStore<StringId> {
   auto Add(llvm::StringRef value) -> StringId {
     auto [it, inserted] = map_.insert({value, StringId(values_.size())});
     if (inserted) {
+      CARBON_CHECK(it->second.index >= 0) << "Too many unique string literals";
```

Comment:
As above.

Reply:
Isn't this the specialization for strings? It seems valuable to have this additional information.

Reply:
It's for strings, not necessarily string literals.

Reply:
Or is this not specific to literals?